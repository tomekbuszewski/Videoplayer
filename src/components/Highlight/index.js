/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-26
 */

import * as React from 'react';
import inject from 'react-jss';

import styles from './index.styles';

type Props = {
  active: boolean,
  caption?: string | boolean,
  image?: string | boolean;
  onClick: Function,
  onMouseEnter: Function,
  onMouseLeave: Function,
  style: Object,
};

const Highlight = (props: Props) => {
  const {
    active,
    caption,
    classes,
    image,
    onMouseEnter,
    onMouseLeave,
    style,
    onClick,
  } = props;

  return (
    <div style={{...style, position: 'absolute' }}>
      <button
        type="button"
        onClick={onClick}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      />
      {active && (
        <React.Fragment>
          {image && <img src={image} alt="" />}
          {caption && <p>{caption}</p>}
        </React.Fragment>
      )}
    </div>
  );
};

Highlight.defaultProps = {
  image: false,
  caption: false,
};

export default inject(styles)(Highlight);
