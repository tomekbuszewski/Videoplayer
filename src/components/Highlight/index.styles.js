import { TRANSITION_SHORTHAND } from '@config/animations';
import { COLOR_BLACK, COLOR_BRAND, COLOR_WHITE } from '@config/colors';
import { FONT_SIZE_MEDIUM } from '@config/fonts';
import {
  BACKDROP,
  STEP_EXTRASMALL,
  STEP_MEDIUM,
  STEP_SMALL,
  GET_STEP,
  OPACITY_LOW,
} from '@config/visuals';

const borderStep = `${GET_STEP(STEP_MEDIUM) / 2}px`;

export default {
  Highlight: {
    position: 'absolute',
    top: 0,
  },

  HighlightRotated: {
    transform: `rotate(180deg) translate(${GET_STEP(STEP_MEDIUM)}px, ${GET_STEP(STEP_MEDIUM)}px)`,

    '& $CaptionHolder': {
      transform: `translate(-50%, ${GET_STEP(STEP_SMALL)}px) rotate(180deg)`,
    },

    '& $CaptionHolderActive': {
      transform: 'translate(-50%, 0) rotate(180deg)',
    },
  },

  Pointer: {
    cursor: 'pointer',
    background: 'none',
    width: 0,
    height: 0,
    borderStyle: 'solid',
    borderWidth: `0 ${borderStep} ${borderStep} ${borderStep}`,
    borderColor: `transparent transparent ${COLOR_BRAND} transparent`,
    transform: 'translateX(-50%)',

    '&:hover': {
      borderColor: `transparent transparent ${COLOR_WHITE} transparent`,
    },
  },

  CaptionHolder: {
    position: 'absolute',
    pointerEvents: 'none',
    minWidth: 200,
    opacity: 0,
    transform: `translate(-50%, ${GET_STEP(STEP_SMALL)}px)`,
    borderRadius: GET_STEP(STEP_EXTRASMALL),
    padding: GET_STEP(STEP_SMALL),
    boxShadow: BACKDROP(STEP_EXTRASMALL, COLOR_BLACK, 0),
    fontSize: FONT_SIZE_MEDIUM,

    background: COLOR_WHITE,

    transition: `${TRANSITION_SHORTHAND('opacity')}, ${TRANSITION_SHORTHAND('transform')}, ${TRANSITION_SHORTHAND('box-shadow', true)}`,
  },

  CaptionHolderActive: {
    opacity: 1,
    height: 'auto',
    transform: 'translate(-50%, 0)',
    boxShadow: BACKDROP(STEP_EXTRASMALL, COLOR_BLACK, OPACITY_LOW),
    pointerEvents: 'all',
  },
};
