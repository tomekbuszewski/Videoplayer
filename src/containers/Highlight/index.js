/**
 * @flow
 * @author tomekbuszewski
 * @since 2018-11-26
 */

import * as React from 'react';

import View from "@src/components/Highlight";

type Props = {
  classes: Object,
  duration: number,
  highlights: Object[],
  position: number,
  onClick: Function,
};

type State = {
  active: boolean,
};

class Highlight extends React.Component<Props, State> {
  state = {
    active: false,
  };

  toggleHover = (): void => {
    const { active } = this.state;

    this.setState({
      active: !active,
    });
  };

  render() {
    const { active } = this.state;

    return (
      <View
        {...this.props}
        active={active}
        onMouseEnter={this.toggleHover}
        onMouseLeave={this.toggleHover}
      />
    );
  }
}

export default Highlight;
