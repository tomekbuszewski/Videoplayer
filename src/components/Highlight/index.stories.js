import * as React from 'react';

import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import Highlight from '.';

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add(
    'Highlight',
    (() => <Highlight
      active
      image="https://www.fillmurray.com/300/200"
      onMouseEnter={() => false}
      onMouseLeave={() => false}
      onClick={() => false}
    />),
  );
