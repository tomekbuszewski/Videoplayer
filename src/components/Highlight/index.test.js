import * as React from 'react';
import renderer from 'react-test-renderer';

import Highlight from '.';

describe('Highlight tests', () => {
  it('should match snapshot with caption and image', () => {
    const component = renderer.create(
      <Highlight
        active
        caption="Test caption"
        image="https://www.fillmurray.com/300/200"
        onMouseEnter={() => false}
        onMouseLeave={() => false}
        onClick={() => false}
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with caption and without image', () => {
    const component = renderer.create(
      <Highlight
        active
        caption="Test caption"
        onMouseEnter={() => false}
        onMouseLeave={() => false}
        onClick={() => false}
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with image and without caption', () => {
    const component = renderer.create(
      <Highlight
        active
        image="https://www.fillmurray.com/300/200"
        onMouseEnter={() => false}
        onMouseLeave={() => false}
        onClick={() => false}
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot without image or caption', () => {
    const component = renderer.create(
      <Highlight
        active
        onMouseEnter={() => false}
        onMouseLeave={() => false}
        onClick={() => false}
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
