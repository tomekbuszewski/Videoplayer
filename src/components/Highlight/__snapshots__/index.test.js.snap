// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight tests should match snapshot with caption and image 1`] = `
<div
  className="Highlight-Highlight-0-1-1 Highlight-HighlightRotated-0-1-2"
  style={Object {}}
>
  <button
    className="Highlight-Pointer-0-1-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
  <div
    className="Highlight-CaptionHolder-0-1-4 Highlight-CaptionHolderActive-0-1-5"
  >
    <img
      alt=""
      src="https://www.fillmurray.com/300/200"
    />
    <p>
      Test caption
    </p>
  </div>
</div>
`;

exports[`Highlight tests should match snapshot with caption and without image 1`] = `
<div
  className="Highlight-Highlight-0-1-1 Highlight-HighlightRotated-0-1-2"
  style={Object {}}
>
  <button
    className="Highlight-Pointer-0-1-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
  <div
    className="Highlight-CaptionHolder-0-1-4 Highlight-CaptionHolderActive-0-1-5"
  >
    <p>
      Test caption
    </p>
  </div>
</div>
`;

exports[`Highlight tests should match snapshot with image and without caption 1`] = `
<div
  className="Highlight-Highlight-0-1-1 Highlight-HighlightRotated-0-1-2"
  style={Object {}}
>
  <button
    className="Highlight-Pointer-0-1-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
  <div
    className="Highlight-CaptionHolder-0-1-4 Highlight-CaptionHolderActive-0-1-5"
  >
    <img
      alt=""
      src="https://www.fillmurray.com/300/200"
    />
  </div>
</div>
`;

exports[`Highlight tests should match snapshot without image or caption 1`] = `
<div
  className="Highlight-Highlight-0-1-1 Highlight-HighlightRotated-0-1-2"
  style={Object {}}
>
  <button
    className="Highlight-Pointer-0-1-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    type="button"
  />
</div>
`;
